lib/SlackBot.js

```diff
@@ -0,0 +1,348 @@
+'use strict';
+
+var WebClient = require('@slack/client').WebClient;
+var octokit = require('@octokit/rest')();
+var YAML = require('yaml');
+var requestPromise = require('request-promise');
+var schedule = require('node-schedule');
+var Cesium = require('cesium');
+var Promise = require('bluebird');
+var fs = require('fs');
+var handlebars = require('handlebars');
+var path = require('path');
+var moment = require('moment');
+var defined = Cesium.defined;
+var RuntimeError = Cesium.RuntimeError;
+
+var dateLog = require('./dateLog');
+
+var SlackBot = {};
+
+var repoNameRegex = /repos\/([\w-]+\/[\w-]+)/;
+
+/** Fetches Slack metadata and authenticates with GitHub. Sets Slackbot.ready to true when done.
+ *
+ * @param {Object} options Object with the following properties:
+ * @param {String} options.token A Slack API token. See this repository's README for instructions on obtaining one.
+ * @param {String} options.configUrl The GitHub API URL to a YAML file containing the release schedule and other SlackBot config.
+ * @param {Object[]} options.repositories Array of repository settings objects.
+ *
+ * @return {undefined}
+ */
+SlackBot.init = function(options) {
+  this._token = options.token;
+  this._configUrl = options.configUrl;
+  this._repositories = options.repositories;
+  this._isDisabled = false;
+  this._userIDs = {};
+  this._userData = {};
+  this._channelIDs = {};
+  this._ready = false;
+  if (!defined(this._token)) {
+    dateLog('No Slack token found. SlackBot is disabled.');
+    this._isDisabled = true;
+    return;
+  }
+
+  this._slackClient = new WebClient(this._token);
+
+  // Authenticate using the first token found.
+  this._authenticateGitHub();
+
+  var that = this;
+  this._getSlackMetadata()
+    .then(function() {
+      that._ready = true;
+    })
+    .catch(function(error) {
+      dateLog('SlackBot._getSlackMetadata failed. SlackBot is disabled. ' + error);
+    });
+};
+
+/**
+ * Sets up scheduled jobs, like weekly statistics and release reminders.
+ *
+ * @return {undefined}
```

Comment:
We should be able to update eslint. The docs and the configuration should match our other repos.

Reply:
I think this might have to be a separate PR. I ran into issues updating, and it's still complaining. I wrote up an issue for it https://github.com/AnalyticalGraphicsInc/cesium-concierge/issues/134.